consensus/safety-rules/src/lib.rs

```diff
@@ -174,9 +174,9 @@ impl SafetyRules {
     /// The update function is invoked whenever a system learns about a potentially high QC.
     pub fn update(&mut self, qc: &QuorumCert) {
         // Preferred block rule: choose the highest 2-chain head.
-        if qc.parent_block_round() > self.state.preferred_block_round() {
+        if qc.vote_data().parent().round() > self.state.preferred_block_round() {
```

Comment:
maybe it's just me but I kind of like the old method because it encapsulates the internal structure of QC very well and is more stable against internal refactors, compared with the proposed way.

Reply:
I like to have some helper functions too, at least sth like `qc.certified_block().id()`, `qc.parent_block().id()`

Reply:
With respect to Lei's concern, sed is a mighty powerful tool to help with refactors to our own code. With that being said, if someone else depends on us, they may need to learn how to use sed as well ;)

Agree with Zekun's proposal, will update.

My concern with exposing proposed is that at this point in time, it is not really proposed but has actually been certified, so Zekun's proposal hides that naming issue.